locksmith/__tests__/fulfillment/tokenTransferApproval.test.ts

```diff
@@ -0,0 +1,55 @@
+import { BigNumber } from 'ethers/utils'
+import TokenTransferApproval from '../../src/fulfillment/tokenTransferApproval'
+
+const nock = require('nock')
+
+nock.back.fixtures = `${__dirname}/fixtures/paymentProcessor`
+nock.back.setMode('lockdown')
+nock.disableNetConnect()
+
+afterAll(() => {
+  nock.restore()
+})
+
+describe('TokenTransferApproval', () => {
+  let provider = 'http://0.0.0.0:8545'
+  let credentials =
+    '0xe5986c22698a3c1eb5f84455895ad6826fbdff7b82dbeee240bad0024469d93a'
+  let lockContractAddress = '0x5Cd3FC283c42B4d5083dbA4a6bE5ac58fC0f0267'
+  let erc20ContractAddress = '0x591ad9066603f5499d12ff4bc207e2f577448c46'
+
+  describe('approve', () => {
+    it('returns a transaction hash for the approval request', async () => {
+      expect.assertions(1)
+      let { nockDone } = await nock.back('approve_token_transfer.json')
+      let transferApprover = new TokenTransferApproval(provider, credentials)
+      let transaction = await transferApprover.approve(
+        '200',
+        lockContractAddress,
+        erc20ContractAddress
+      )
+      expect(transaction).toEqual(
+        expect.objectContaining({
+          chainId: 1984,
+          data:
```

Comment:
is there any way this can be generated from an ethers.utils.Interface() instead of being impenetrable hex? if not, I suppose it's fine, but the test will fail with the smallest smart contract change, and it could be hard to verify that the difference is because of this and not a bug